src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs

```diff
@@ -322,8 +321,21 @@ public async Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestP
 
             if (arcadeItem != null && repoUri != arcadeItem.RepoUri)
             {
-                // Files in arcade repository
-                List<GitFile> engCommonFiles = await GetCommonScriptFilesAsync(arcadeItem.RepoUri, arcadeItem.Commit);
+                // Files in arcade repository. All Arcade items have a GitHub repo URI by default so we need to change the
+                // URI from we are getting the eng/common files. If in an AzDO context we change the URI to that of
+                // dotnet-arcade in dnceng
+
+                string engCommonRepoUri = arcadeItem.RepoUri;
+
+                if (Uri.TryCreate(repoUri, UriKind.Absolute, out Uri parsedUri))
+                {
+                    if (parsedUri.Host == "dev.azure.com" || parsedUri.Host.EndsWith("visualstudio.com"))
```

Comment:
We shouldn't hardcode these.  Instead,. we should pass a RemoteFactory through to this method, which can then create a github remote to get the files.  See the dependency graph creator for an example on how this is used.